### PR TITLE
docs(framework:skip) Add docs for distributed Docker Compose guide

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

- Add docs for distributed Docker Compose guide.
- Depends on: #3916, #3924 

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+proceeding with this guide.
+
+Prerequisites
+-------------
+
+Before you begin, make sure you have the following prerequisites:
+
+- The ``flwr`` CLI is :doc:`installed <../how-to-install-flower>` locally.
+- The Docker daemon is running on your local machine and the remote machine.
+- Docker Compose is installed on both your local machine and the remote machine.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:20]:
```suggestion
- Docker Compose V2 is installed on both your local machine and the remote machine.
```

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+   .. code-block:: bash
+
+      $ git clone --depth=1 https://github.com/adap/flower.git
+      $ cd flower/src/docker/distributed
+
+#. Get the IP address from the remote machine and save it for later.
+
+#. If you don't have any certificates, you can generate self-signed certificates using the ``certs.yml``
+   Compose file. If you have certificates, you can continue with Step 2.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:43]:
```suggestion
#. Use the ``certs.yml` Compose file to generate your own self-signed certificates.
   If you have certificates, you can continue with Step 2.
```

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 3]
```diff
@@ -0,0 +1,158 @@
+Deploy Flower on Multiple Machines with Docker Compose
+======================================================
+
+This guide will help you set up a Flower project on multiple machines using Docker Compose.
+
+You will learn how to run the Flower client and server components on two separate machines,
+with Flower configured to use TLS encryption and persist SuperLink state across restarts.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:7]:
```suggestion
with Flower configured to use TLS encryption and persist SuperLink state across restarts. A server consists of a SuperLink and ``SuperExec``. For more details
about the Flower architecture, refer to the :doc:Flower Architecture`../explanation-flower-architecture.rst` explainer page.
```

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+------------------------------------------
+
+Log into the remote machine using ``ssh`` and run the following command to start the
+SuperLink and SuperExec services:
+
+.. code-block:: bash
+
+   $ ssh remote
+   $ export PROJECT_DIR=../quickstart-sklearn-tabular
+   $ docker compose -f server/compose.yml up --build -d
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:93]:
```suggestion
   $ ssh <your-remote-machine>
   # In your remote machine
   $ cd <path-to-``server``-directory>
   $ export PROJECT_DIR=../quickstart-sklearn-tabular
   $ docker compose -f server/compose.yml up --build -d
```

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+Go back to your terminal on your local machine.
+
+Step 4: Start the Flower Client Components
+------------------------------------------
+
+On your local machine, run the following command to start the client components:
+
+.. code-block:: bash
+
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:108]:
```suggestion

   # In the `docker/distributed/client` directory
```

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+.. note::
+
+   The Path of the ``PROJECT_DIR`` should be relative to the location of the ``client`` Docker
+   Compose files.
+
+Step 5: Run Your Flower Project
+-------------------------------
+
+Specify the remote SuperExec IP addresses and the path to the root certificate in the
+``pyproject.toml`` file:
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:121]:
```suggestion
``[tool.flwr.federations.remote-superexec]`` table in the ``pyproject.toml`` file. Here, we have named our remote federation ``remote-superexec``:
```

**Review by chongshenng — CHANGES_REQUESTED** [review]
Thanks for the PR @Robert-Steiner! Overall looks good. I've left a few comments in the PR itself. 

Nits: 
* Can you use code-block for ``SuperExec`` throughout? 
* Can you format and lint the code with ``rstfmt``? We'll be using this in our CI in the coming days, so it'll be good to get it formatted as well.

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+Shut down the Flower server components and delete the SuperLink state:
+
+.. code-block:: bash
+
+   $ ssh <your-remote-machine>
+   $ cd <path-to-``distributed``-directory>
+   $ docker compose -f server/compose.yml down -v
+
+.. |quickstart_skearn_tabular| replace::
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:166]:
```suggestion
.. |quickstart_sklearn_tabular| replace::
```

File: doc/source/docker/tutorial-deploy-on-multiple-machines.rst [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+   ``examples/quickstart-sklearn-tabular``
+
+.. _quickstart_skearn_tabular: https://github.com/adap/flower/tree/main/examples/quickstart-sklearn-tabular
```
Comment (chongshenng) [inline doc/source/docker/tutorial-deploy-on-multiple-machines.rst:170]:
```suggestion
.. _quickstart_sklearn_tabular: https://github.com/adap/flower/tree/main/examples/quickstart-sklearn-tabular
```

**Review by chongshenng — APPROVED** [review]
lgtm!

**Review by tanertopal — APPROVED** [review]
Lgtm!

**Pull request merged**